crates/miden-tx/src/tests/kernel_tests/test_fpi.rs

```diff
@@ -839,40 +879,339 @@ fn test_simple_nested_fpi() {
+            )
+            .unwrap()
+            .with_supports_all_types();
+
+            let foreign_account = AccountBuilder::new(ChaCha20Rng::from_os_rng().random())
+                .with_component(foreign_account_component)
+                .build_existing()
+                .unwrap();
+
+            foreign_accounts.push(foreign_account)
+        }
+
+        // ------ NATIVE ACCOUNT ---------------------------------------------------------------
+        let native_account = AccountBuilder::new(ChaCha20Rng::from_os_rng().random())
+            .with_component(
+                AccountMockComponent::new_with_slots(TransactionKernel::testing_assembler(), vec![])
+                    .unwrap(),
+            )
+            .build_existing()
+            .unwrap();
+
+        let mut mock_chain = MockChain::with_accounts(&[
+            vec![native_account.clone()], foreign_accounts.clone()
+        ].concat());
+
+        mock_chain.seal_block(None, None);
+
+        let advice_inputs =
+            get_mock_fpi_adv_inputs(foreign_accounts.iter().collect::<Vec<&Account>>(), &mock_chain);
+
+        let code = format!(
+            "
+            use.std::sys
+
+            use.miden::tx
+
+            begin
+                # get the storage item at index 0
+                # pad the stack for the `execute_foreign_procedure` execution
+                padw padw padw push.0.0
+                # => [pad(14)]
+
+                # push the index of desired storage item
+                push.0
+
+                # get the hash of the `get_item` account procedure
+                push.{foreign_account_proc_hash}
+
+                # push the foreign account ID
+                push.{foreign_suffix}.{foreign_prefix}
+                # => [foreign_account_id_prefix, foreign_account_id_suffix, FOREIGN_PROC_ROOT, storage_item_index, pad(14)]
+
+                exec.tx::execute_foreign_procedure
+                # => [storage_value]
+
+                exec.sys::truncate_stack
+            end
+            ",
+            foreign_account_proc_hash = foreign_accounts.last().unwrap().code().procedures()[0].mast_root(),
+            foreign_prefix = foreign_accounts.last().unwrap().id().prefix().as_felt(),
+            foreign_suffix = foreign_accounts.last().unwrap().id().suffix(),
+        );
+
+        let tx_script = TransactionScript::compile(
+            code,
+            vec![],
+            TransactionKernel::testing_assembler().with_debug_mode(true),
+        )
+        .unwrap();
+
+        let tx_context = mock_chain
+            .build_tx_context(native_account.id(), &[], &[])
+            .advice_inputs(advice_inputs.clone())
+            .tx_script(tx_script)
+            .build();
+
+        let block_ref = tx_context.tx_inputs().block_header().block_num();
+        let note_ids = tx_context
+            .tx_inputs()
+            .input_notes()
+            .iter()
+            .map(|note| note.id())
+            .collect::<Vec<_>>();
```

Comment:
We don't use those notes in the test so let's remove them to improve readability.

We have this line a few times in the FPI tests but I think we never actually use notes, can you double check and remove if they're not used?